src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    @Override
+    public List<Machine> get() {
+        return List.of(new Truck(), new Truck());
+    }
```

Comment:
Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created. The current implementation allows creating producers for any `Machine` type, which is incorrect. Change the return type of the `get()` method to `List<Truck>` to ensure that only `Truck` objects can be produced.